runner.go

```diff
@@ -64,8 +72,21 @@ func (r *Runner) RunChecksumCalculator(fileNamePatterns []string) int {
 }
 
 func (r *Runner) doRun() {
-	ignoresNew := detector.ReadConfigFromRCFile(readRepoFile())
-	detector.DefaultChain().Test(r.additions, ignoresNew, r.results)
+	rcConfigIgnores := detector.ReadConfigFromRCFile(readRepoFile())
+	scopeMap := readScopeConfig()
+	additionsToScan := detector.IgnoreAdditionsByScope(r.additions, rcConfigIgnores, scopeMap);
+	detector.DefaultChain().Test(additionsToScan, rcConfigIgnores, r.results)
+}
+
+func readScopeConfig() map[string][]string {
+	var scope map[string][]string
+	err := yaml.Unmarshal([]byte(scopeConfig), &scope)
+	if err != nil {
+		log.Println("Unable to parse scope_config.go")
```

Comment:
Why would parsing a hard-coded string fail ?
Why not store the scope config object directly instead of parsing it every time ?
The error message is a little misleading too.

Reply:
Moved the config to a object instead of parsing